enhancements/bpfd-on-ocp.md

```diff
@@ -0,0 +1,424 @@
+a Kubernetes cluster.
+
+The benefits of this solution include the following:
+- Improved security because only the bpfd daemon, which can be tightly
+  controlled, requires the privileges needed to load BPF programs, while access
+  to the API can be controlled via standard RBAC methods. This could also allow
+  us to remove CAP_BPF from the privileged container defaults.
+- Improved ease of use and productivity because all developers use a standard
+  framework for managing BPF programs which reduces duplicated efforts and the
+  learning curve.
+- Support for the coexistence of multiple BPF programs from multiple users.
+- Improved visibility into what BPF programs are running on a system, which
+  enhances the debuggability for developers, admins, and customer support.
+
+### User Stories
+
+1. As an OCP developer, I would like to have a consistent solution for loading,
+   managing, and debugging across all of our BPF-enabled applications and
+   components.
+
+   Without a common solution, each application must solve many of these problems
+   on its own, which leads to duplicated code, functionality, and/or different
+   solutions. For example, we currently have three different OCP applications
+   (Ingress Node Firewall, Network Observability, and ACS) that use different
+   methods of loading and managing BPF programs. A common framework can also
+   simplify the debugging of interactions between BPF applications and make it
+   easier for developers to work on multiple projects.
+
+2. As an OCP Developer/product manager, I would like to introduce functionality
+   in a future release that uses BPF programs and have these new features
+   interact in a predictable way with existing functionality, other operators,
+   and local customizations.
+
+   BPF allows low-level features to be programmed much more quickly than via
+   traditional methods, e.g submitting kernel patches. Developers should be able
+   to easily ship new functionality which makes use of BPF in Openshift without
+   worrying about clobbering other developers who intend to make use of similar
+   hook points and BPF programs. 
+
+3. As an OCP Customer, I want to deploy BPF programs in a supported way. 
+
+   As an OCP customer, I want to work around an OCP/RHEL bug by using a
+   MachineConfig (or something similar) to deploy a small BPF program to every
+   node in my cluster, without voiding my warranty.
+
+   As an OCP customer, I want to add functionality to OCP/RHEL by using a
+   MachineConfig (or something similar) to deploy a small (or not-so-small) BPF
+   program to every node in my cluster, without voiding my warranty.
+
+   (E.g., there is a customer who wants to use BPF to implement OVS-style SLB
+   Bonding on a kernel bond interface, using BPF. The upstream Linux maintainer
+   rejected a patch to implement it in the kernel driver directly, in part
+   because it would be so easy to implement it with BPF.)
```

Comment:
Would we make users resort to BPF directly for that? Or would we provide an API and controller that would generate the BPF for them?